common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -507,6 +552,154 @@ bool base_wlan_hal_dwpal::attach_ctrl_interface(int vap_id)
+    static __thread std::function<DWPAL_Ret(char *ifname, int event, int subevent, size_t len,
+                                            unsigned char *data)>
+        nl_handler_cb_wrapper;
+    nl_handler_cb_wrapper = [&](char *ifname, int event, int subevent, size_t len,
+                                unsigned char *data) -> DWPAL_Ret {
+        LOG(DEBUG) << __func__ << " ifname = " << std::string(ifname) << " event = " << event
+                   << " subevent = " << subevent << " len = " << len;
+
+        if ((len == 0) || (data == NULL)) {
+            LOG(DEBUG) << __func__ << "len=0 and/or data is NULL ==> Abort!\n";
+            return DWPAL_FAILURE;
+        }
+        if (event == NL80211_CMD_VENDOR) {
+            if (len >= max_buffer_size) {
+                LOG(DEBUG) << __func__ << "NL size exceeds out_buffer size ==> Abort!\n";
+                return DWPAL_FAILURE;
+            }
+            // copy get result from nl buffer (data[i])
+            for (int i = 0; i < (int)len; i++) {
+                out_buffer[i] = data[i];
+            }
+            *out_buffer_size = len;
+
+        } else {
+            LOG(DEBUG) << __func__ << "non vendor event not handled ";
+            return DWPAL_FAILURE;
+        }
+        return DWPAL_SUCCESS;
+    };
+    auto nl_handler_cb = [](char *ifname, int event, int subevent, size_t len,
+                            unsigned char *data) -> DWPAL_Ret {
+        return nl_handler_cb_wrapper(ifname, event, subevent, len, data);
+    };
+
+    while (*out_buffer_size == 0) { /* wait for reply with valid data */
+
+        //parsing will be done in callback func
+        if (dwpal_driver_nl_msg_get(m_dwpal_nl_ctx, DWPAL_NL_SOLICITED_EVENT, nl_handler_cb,
+                                    NULL) == DWPAL_FAILURE) {
+            LOG(ERROR) << " dwpal_driver_nl_msg_get failed,"
+                       << " ctx=" << m_dwpal_nl_ctx;
+            detach();
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool base_wlan_hal_dwpal::dwpal_nl_cmd_scan_dump()
+{
+    bool err = true;
+    // Passing a lambda with capture is not supported for standard C function
+    // pointers. As a workaround, we create a static (but thread local) wrapper
+    // function that calls the capturing lambda function.
+    static __thread std::function<DWPAL_Ret(struct nl_msg * msg)> nl_handler_cb_wrapper;
+    nl_handler_cb_wrapper = [&](struct nl_msg *msg) -> DWPAL_Ret {
+        if (process_dwpal_nl_event(msg) != true) {
+            LOG(ERROR) << "User's netlink handler function failed!";
+            return DWPAL_FAILURE;
+        }
+        return DWPAL_SUCCESS;
+    };
+    auto nl_handler_cb = [](struct nl_msg *msg) -> DWPAL_Ret { return nl_handler_cb_wrapper(msg); };
+
+    if (dwpal_driver_nl_scan_dump(m_dwpal_nl_ctx, (char *)m_radio_info.iface_name.c_str(),
+                                  nl_handler_cb) != DWPAL_SUCCESS) {
+        err = false;
```

Comment:
add `LOG(ERROR)` and return false

Reply:
changed it and removed the err since it has no other use